doc/metadata_routing.rst

```diff
@@ -116,26 +132,23 @@ that :func:`~model_selection.cross_validate` does not pass the weights along::
   ...     scoring=weighted_acc,
   ... )
 
-If :meth:`linear_model.LogisticRegressionCV.set_fit_request` has not
-been called, :func:`~model_selection.cross_validate` will raise an
-error because ``sample_weight`` is passed in but
-:class:`~linear_model.LogisticRegressionCV` would not be explicitly configured
-to recognize the weights.
+If :meth:`linear_model.LogisticRegressionCV.set_fit_request` had not been called,
+:func:`~model_selection.cross_validate` would raise an error because ``sample_weight``
+is passed but :class:`~linear_model.LogisticRegressionCV` would not be explicitly
+configured to recognize the weights.
```

Comment:
That's an interesting suggestion, feel free to open a separate issue for it.

Reply:
Well, it seems that the error message is already stating this information, except for `cross_validate` for specific reasons of it's implementation. Here, it gets substituted by a changed version:

sklearn/model_selection/_validation.py
```python
        except UnsetMetadataPassedError as e:
            # The default exception would mention `fit` since in the above
            # `process_routing` code, we pass `fit` as the caller. However,
            # the user is not calling `fit` directly, so we change the message
            # to make it more suitable for this case.
            raise UnsetMetadataPassedError(
                message=(
                    f"{sorted(e.unrequested_params.keys())} are passed to cross"
                    " validation but are not explicitly requested or unrequested. See"
                    " the Metadata Routing User guide"
                    " <https://scikit-learn.org/stable/metadata_routing.html> for more"
                    " information."
                ),
                unrequested_params=e.unrequested_params,
                routed_params=e.routed_params,
            )
```
I don't think this is so easy to change.

Maybe, I could modify the error message that is checked in `sklearn/tests/test_metaestimators_metadata_routing.py,` though, because it's only considering the usual case...

Reply:
That test file only checks for meta-estimators, and `cross_validate` is not a meta estimator. But I'm happy to make sure `cross_validate` also raises a nice error.